policyengine_us/variables/gov/states/az/tax/income/deductions/itemized/az_itemized_deduction.py

```diff
@@ -0,0 +1,72 @@
+from policyengine_us.model_api import *
+
+
+class az_itemized_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Arizona Itemized Deductions"
+    unit = USD
+    documentation = "Arizona Form 140 Schedule A"
+    definition_period = YEAR
+    defined_for = StateCode.AZ
+    reference = (
+        "https://law.justia.com/codes/arizona/2022/title-43/section-43-1042/",
+        "https://azdor.gov/forms/individual/itemized-deduction-adjustments-form",
+        "https://azdor.gov/forms/individual/form-140-resident-personal-income-tax-form-calculating",
+    )
+    definition_period = YEAR
+    defined_for = StateCode.AZ
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.irs.deductions
+        deductions = [
+            deduction
+            for deduction in p.itemized_deductions
+            if deduction
+            not in [
+                "salt_deduction",
+                "medical_expense_deduction",
+                "charitable_deduction",
+            ]
+        ]
+        federal_deductions = add(tax_unit, period, deductions)
+        # Adjustment to Medical and Dental Expresses
+        medical_expense = add(tax_unit, period, ["medical_expense"])
+        medical_expense_deduction = tax_unit(
+            "medical_expense_deduction", period
+        )
+        az_medical_expense_deduction = where(
+            medical_expense >= medical_expense_deduction,
+            medical_expense - medical_expense_deduction,
+            medical_expense_deduction - medical_expense,
+        )
+        # charitable_deduction = tax_unit("az_charitable_deduction", period)
+
+        # Adjustment to Interest Deduction
+        # mortgage_interest = the amount of mortgage interest you paid for 2022 that is equal to
+        # the amount of federal credit, if you received a federal credit for interest paid on
+        # mortgage credit certificates (from federal Form 8396).
+
+        # Adjustments to Charitable Contributions
+        # Amount of charitable contributions for which you are claiming a credit under Arizona law
+        charitable_deduction = tax_unit("charitable_deduction", period)
+        charitable_contributions_credit = tax_unit(
+            "az_charitable_contributions_credit", period
+        )
+
+        charitable_deduction_allowed = where(
```

Comment:
filers can still take a deduction for the amount not claimed as a credit.

since the credit will always be more valuable, we can assume they take the credit for the maximum amount allowable, then the deduction on any remainder